src/js/controller/controller.js

```diff
@@ -859,7 +859,9 @@ Object.assign(Controller.prototype, {
                     // This prevents media from replaying when a completed video scrolls into view
                     _model.set('playOnViewable', false);
                     _model.set('state', STATE_COMPLETE);
-                    _this.trigger(PLAYLIST_COMPLETE, {});
+                    if (!_model.get('related') || _model.get('item') === _model.get('playlist').length - 1) {
```

Comment:
What if it's a regular playlist that hasn't completed and it's there's no recs (no related)?

Reply:
If it's a regular playlist, the `complete` event fires but the `playlistComplete` will not fire until the regular playlist finishes its last item. I can see this behavior occurring locally rn with my changes.